lstchain/scripts/lstchain_r0_to_r0g.py

```diff
@@ -0,0 +1,206 @@
+            stream.move_to_new_table("DataStream")
+            stream.write_message(input_streams[i].DataStream[0])
+
+            stream.move_to_new_table("Events")
+
+            for event in input_streams[i].Events:
+                # skip corrupted events:
+                if event.event_id == 0:
+                    continue
+                # Get the event type, from the first loop over the files:
+                evtype = event_type_val[event_id==event.event_id][0]
+
+                if ((evtype == EventType.SUBARRAY.value) |
+                    (evtype == EventType.UNKNOWN.value)):
+                    # Find pixels with HG above gain switch threshold:
+                    wf = protozfits.any_array_to_numpy(event.waveform)
+                    num_gains = int(wf.size / num_pixels / num_samples)
+
+                    if num_gains != 2:
+                        logging.error('You are attempting gain selection on '
+                                      'data with only one gain!')
+                        exit(1)
+
+                    wf = wf.reshape((num_gains, num_pixels, num_samples))
+                    hg = wf[0]
+                    lg = wf[1]
+                    use_lg = (np.max(hg[:, SAMPLE_START:SAMPLE_END], axis=1)
+                              > THRESHOLD)
+                    new_wf = np.where(use_lg[:, None], lg, hg) # gain-selected
+                    event.waveform.data = new_wf.tobytes()
+                    event.num_channels = 1  # Just one gain stored!
+
+                    pixel_status = protozfits.any_array_to_numpy(event.pixel_status)
+                    # Set to 0 the status bit of the removed gain:
+                    new_status = np.where(use_lg,
+                                          pixel_status & 0b1011,
+                                          pixel_status & 0b0111)
+                    event.pixel_status.data = new_status.tobytes()
+
+                stream.write_message(event)
+
+            stream.close()
+            input_streams[i].close()
+
+    logging.info('R0 to R0G conversion finished successfully!')
+    return(0)
+
+
+def get_event_types(input_file):
+
+    # For heuristic flat field identification (values refer to
+    # baseline-subtracted HG integrated waveforms):
+    MIN_FLATFIELD_ADC = 3000
+    MAX_FLATFIELD_ADC = 12000
+    MIN_FLATFIELD_PIXEL_FRACTION = 0.8
+
+    standard_config['source_config']['LSTEventSource'][
+        'apply_drs4_corrections'] = False
+    standard_config['source_config']['LSTEventSource'][
+        'pointing_information'] = False
+
+    event_type = []
+    event_id = []
+    with EventSource(input_url=input_file,
+                     config=Config(standard_config['source_config'])) as source:
+        source.pointing_information = False
+        source.trigger_information = True
+        source.log.setLevel(logging.WARNING)
+        try:
+            for event in source:
+                if event.r0.tel[1].waveform is None:
```

Comment:
EVBv6 data will have always the waveforms in `r1`.

Reply:
?? I just tried, and when no drs4 calib is applied the r0 is filled by the source (also r1), so I think it works.  But ok, the baseline might eventually be different, so I will ~~either use r1 or~~ subtract the correct one (should be equivalent, right?). On second thought, using r1 with no calibration is dangerous because it might occasionally take the low gain, and the condition is on adc counts above baseline.

Reply:
> I just tried, and when no drs4 calib is applied the r0 is filled by the source (also r1), so I think it works. 

If no pe calibration is performed, the original waveform is also filled into r0. 

I didn't implement storing the unscaled waveform in case of pe calibration alrady applied, since that is relatively meaningless... it's not original adc counts but scaled pe.

Reply:
So no issue here, right?